go/teams/seitan_v2.go

```diff
@@ -0,0 +1,276 @@
+package teams
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/base64"
+	"errors"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/go-crypto/ed25519"
+	"golang.org/x/crypto/nacl/secretbox"
+	"golang.org/x/net/context"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+// Seitan tokens v2 have a '+' as the sixth character. We use this
+// to distinguish from email invite tokens (and team names).
+// See `IsSeitanyV2`
+const seitanEncodedIKeyV2PlusOffset = 6
+
+// "Invite Key Version 2"
+type SeitanIKeyV2 string
+
+// "Packed Encrypted Invite Key Version 2"
+// All following 3 structs should be considered one. When any changes,
+// Version in PEPubKey has to be bumped up.
+type SeitanPEPubKey struct {
+	_struct              bool `codec:",toarray"`
+	Version              uint
+	TeamKeyGeneration    keybase1.PerTeamKeyGeneration
+	RandomNonce          keybase1.BoxNonce
+	EncryptedKeyAndLabel []byte // keybase1.SeitanPubKeyAndLabel MsgPacked and encrypted
+}
```

Comment:
I'm wondering if we can use the same struct for both seitans

Reply:
yes this is better